packages/marketplace/src/components/ui/developer-confirm-subscription.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import {
+  selectCreateDeveloperSubscriptionLoading,
+  selectCreatedDeveloperSubscription,
+} from '@/selector/developer-subscriptions'
+import { DeveloperModel } from '@reapit/foundations-ts-definitions'
+import { Button, Modal, ModalProps, SubTitleH6 } from '@reapit/elements'
+import { developerCreateSubscription } from '@/actions/developer-subscriptions'
+import linkStyles from '@/styles/elements/link.scss?mod'
+
+export interface DeveloperConfirmSubscriptionProps extends ModalProps {
+  developer: DeveloperModel
+}
+
+export const handleSubmit = (dispatch, developer: DeveloperModel) => () => {
+  dispatch(
+    developerCreateSubscription({
+      developerId: developer.id || '',
+      user: developer.email || '',
+      applicationId: '', // TBC
+      type: 'developerEdition',
+    }),
+  )
+}
+
+const DeveloperConfirmSubscription: React.FC<DeveloperConfirmSubscriptionProps> = ({
+  visible,
+  title = 'Success',
+  tapOutsideToDissmiss = false,
+  developer,
+  afterClose,
+  ...rest
+}) => {
+  const dispatch = useDispatch()
+  const loading = useSelector(selectCreateDeveloperSubscriptionLoading)
+  const subscription = useSelector(selectCreatedDeveloperSubscription)
+
+  React.useEffect(() => {
+    if (!loading && subscription && afterClose) {
+      afterClose()
```

Comment:
why do you need to run afterClose, so if you want to turn off modal please setState visible to false